sdk/storage/storage-file-datalake/review/storage-file-datalake.api.md

```diff
@@ -1214,13 +1133,7 @@ export interface ListPathsSegmentOptions extends ListPathsOptions {
 }
 
 // @public
-export type ListPathsSegmentResponse = FileSystemListPathsHeaders & PathListModel & {
-    _response: coreHttp.HttpResponse & {
-        bodyAsText: string;
-        parsedBody: PathListModel;
-        parsedHeaders: FileSystemListPathsHeaders;
-    };
-};
+export type ListPathsSegmentResponse = FileSystemListPathsHeaders & PathListModel;
```

Comment:
Would this interface miss row response?

Reply:
Looking into this, it seems that `ListPathsSegmentResponse` was not actually used anywhere in the public API surface, presumably because only `FileSystemListPathsResponse` is used instead. I've removed this unnecessary type.

I also fixed up the typings where `_response` wasn't being typed correctly for headers-only response models, namely:

PathCreateResponse
PathDeleteResponse
FileFlushResponse
FileUploadResponse
PathSetAccessControlResponse
PathSetPermissionsResponse